rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -557,14 +557,15 @@ void SPUThread::do_dma_transfer(const spu_mfc_cmd& args, bool from_mfc)
 
 	if (spu)
 	{
-		for (auto& func : spu->compiled_functions)
+		for (u32 i = 0; i < spu->next_compiled_func_index; i++)
 		{
-			auto faddr = func->contents->addr;
-			auto fsize = func->contents->size;
+			auto& func = spu->compiled_functions[i];
+			auto faddr = func.contents->addr;
+			auto fsize = func.contents->size;
 
 			if (faddr >= eal && faddr + fsize < eal + args.size)
```

Comment:
This check doesn't catch every instance of SPU code being modified by a DMA, changing it to `if (args.eal >= faddr && args.eal <= (faddr + fsize))` fixes RDR hanging at the intro.